platform/ui/src/contextProviders/ViewportGridProvider.tsx

```diff
@@ -46,6 +56,7 @@ const reuseViewport = (idSet, viewport, stateViewports) => {
   const oldIds = {};
   for (const oldViewport of stateViewports) {
     const { viewportId: oldId } = oldViewport;
+    if (!oldId) continue;
```

Comment:
is there a scenario where there is no viewportId? I don't think so, when?

Reply:
Yes, when a viewport has no matching contents it doesn't necessarily get a viewportId

Reply:
So if it is possible that a viewport gets no id, then isn't the fact that we are using the `viewportId` as a key for a `ViewportPane` in `ViewportGrid.tsx` problematic?

Reply:
Yes, it probably is an issue - maybe this doesn't happen any longer, but I'd rather leave the check in as it will create a viewportId for any missing viewport Ids.

Reply:
Sorry I am a bit confused. Is it still possible for a Viewport NOT to have a viewportId?

Reply:
AFAIK it cannot happen, but I prefer to have a check there just in case.

Reply:
Just to be clear. I agree with leaving the check. I was more concerned with the use of `viewportId` as a key for a component.